app/routes.py

```diff
@@ -1,2 +1,333 @@
+            "task": {
+            "id": task.task_id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False if task.completed_at == None else True
+        }
+        }), 200
+
+@tasks_bp.route("/<task_id>",methods=["PUT"])
+def update_task(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+
+    updated_body = request.get_json()
+
+    if task == None:
+        return jsonify(None), 404
+
+    else: 
+        task.title = updated_body["title"]
+        task.description = updated_body["description"]
+        
+        db.session.commit()
+        
+        task_dict = {}
+        task_dict = {
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at is not None
+            }
+        }
+        
+        return jsonify(task_dict), 200
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+    
+    if task == None:
+        return jsonify(None), 404
+
+    else: 
+        db.session.delete(task)
+        db.session.commit()
+    
+    response_body = {}
+    response_body = {
+        "details": f'Task {task.task_id} "{task.title}" successfully deleted'}
+    return jsonify(response_body), 200
+
+def slack_bot(title):
+    query_path = {'channel':'slack_api_test_channel', 'text': title}
+    header = {'Authorization': os.environ.get('BOT')}
+    response = requests.post('https://slack.com/api/chat.postMessage', params = query_path, headers = header)
+    return response.json()
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def update_complete(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+
+    if task == None:
+        return jsonify(None), 404
+
+    task.completed_at = datetime.now()
+    db.session.commit()
+
+    slack_bot(task.title)
+
+    task_dict = {}
+    task_dict = {
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at is not None
+            }
+        }
+    
+    return jsonify(task_dict), 200
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def update_incomplete(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+
+    if task == None:
+        return jsonify(None), 404
+    
+    task.completed_at = None
+
+    db.session.commit()
+
+    task_dict = {}
+    task_dict = {
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at is not None
+            }
+        }
+    
+    return jsonify(task_dict), 200
+
```

Comment:
Great work on the task routes! We can make `routes.py` much shorter if we separate the task and goal routes into separate files